tasks/03_pages/task.py

```diff
@@ -0,0 +1,56 @@
+import concurrent.futures
+import logging
+import sys
+
+from common import sql
+import scraper
+
+
+# set up the log
+logging.basicConfig(filename='scrape.log', filemode='a', encoding='utf-8', level=logging.INFO)
+
+def main():
+    # use the db's `cases` and `pages` table to get a list of pages to scrape
+    with sql.db_cnx() as cnx:
+        c = cnx.cursor()
+        print('Fetching cases left to scrape.')
+        cases = scraper.case_pages_to_fetch(cursor=c)
+        if len(cases) == 0:
+            c.close()
+            
+            
+            print('Scrape completed.')
+            return 
+
+        print('...remaining to scrape:', len(cases))
+    c.close()
+    cnx.close()
+
+    # run the scraper. can tune the `max_workers` for multi-threading.
+    # note that each thread/worker opens and closes a connection to the db.
+    try:
```

Comment:
just a comment on the last point: i'm struggling to understand how to shutdown the executor properly during the keyboard interrupt since the script doesn't exit out after a keyboard interrupt.